tests/python_package_test/test_engine.py

```diff
@@ -2102,7 +2102,7 @@ def test_default_objective_and_metric():
 
 @pytest.mark.skipif(psutil.virtual_memory().available / 1024 / 1024 / 1024 < 3, reason='not enough RAM')
 def test_model_size():
-    X, y = load_boston(return_X_y=True)
+    X, y = make_synthetic_regression()
```

Comment:
I checked all CI jobs and there are no new skips compared to `master` due to dataset replacement.